express-notes-api/server.ts

```diff
@@ -0,0 +1,120 @@
+import express from 'express';
+import { readFile, writeFile } from 'node:fs/promises';
+
+const app = express();
+
+type Entry = {
+  id: number;
+  content: string;
+};
+
+type Data = {
+  nextId: number;
+  notes: {
+    [key: number]: Entry;
+  };
+};
+
+async function write(obj: Data) {
+  const createData = JSON.stringify(obj, null, 2);
+  await writeFile('./data.json', `${createData} \n`);
+}
+
+app.use(express.json());
+
+app.get('/api/notes', async (req, res) => {
+  try {
+    const jsonData = await readFile('./data.json', 'utf-8');
+    const dataParsed: Data = JSON.parse(jsonData);
+    const entryArray: Entry[] = [];
+    for (const entry in dataParsed.notes) {
+      entryArray.push(dataParsed.notes[entry]);
+    }
+    res.json(entryArray);
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ error: 'an unexpected error occurred' });
+  }
+});
+
+app.get('/api/notes/:id', async (req, res) => {
+  try {
+    const jsonData = await readFile('./data.json', 'utf-8');
+    const dataParsed: Data = JSON.parse(jsonData);
+    const id = Number(req.params.id);
+    if (id <= 0 || isNaN(id)) {
```

Comment:
You probably want to check `isNan` first so that you don't compare a non-number against 0.